### PR TITLE
Fix reading Delta Table metadata from Glue

### PR DESCRIPTION
**Pull request by erichwang**
Fixes: https://github.com/trinodb/trino/issues/12147

Caused by this PR:
https://github.com/trinodb/trino/pull/12130

### CODE REVIEW

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Test PR: #12172 
(cc @ilfrin @nineinchnick for potentially streamlining this)

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java [hunk 1]
```diff
@@ -65,8 +65,10 @@ public Optional<String> getOptionalLocation()
 
     public String getLocation()
     {
-        // Default getter requires location to be set. Location is not set only in rare case when in CREATE TABLE flow in Hive connector when
-        // delegate-transactional-managed-table-location-to-metastore is set to true and location is determined by HMS.
+        // Default getter requires location to be set. Location is not set in the following scenarios:
+        // 1. CREATE TABLE flow in Hive connector sets delegate-transactional-managed-table-location-to-metastore to true and location is determined by HMS.
+        // 2. Iceberg format tables
+        // 3. Delta format tables (sometimes)
```
Comment (alexjo2144) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java:71]: I double checked this today. Looks like we always expect the path to be in the storage descriptor. We could change that for Managed tables, but right now it's required.

**Comment by findepi** [comment]
CI https://github.com/trinodb/trino/issues/12151

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java (hunk 1 above)
Comment (erichwang) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java:71]: @alexjo2144, do you want me to remove the assumption that storage won't be specified for delta tables then?

**Comment by erichwang** [comment]
@findepi, it looks like the CI build with secrets failed on something that is transient. Can you kick it again and see if it passes?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java (hunk 1 above)
Comment (alexjo2144) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/Storage.java:71]: I think this change is fine, I made an Issue for follow up https://github.com/trinodb/trino/issues/12180

**Comment by erichwang** [comment]
Does anyone else have any concerns here?

**Review by dain — APPROVED** [review]

**Pull request merged**